framework/aster-frame/src/sync/wait.rs

```diff
@@ -141,7 +115,8 @@ impl WaitQueue {
 ///
 /// By definition, a waiter belongs to the current thread, so it cannot be sent to another thread
 /// and its reference cannot be shared between threads.
-struct Waiter {
+#[doc(hidden)]
+pub struct Waiter {
```

Comment:
Well. In https://github.com/asterinas/asterinas/pull/721#discussion_r1571961761, I think the implementation detail is that "`WaitQueue` is implemented by `Wait` and `Waker`". But `Wait` and `Waker` themselves are not necessarily the implementation detail.

@StevenJiang1110 Do I understand your comments correctly?

Reply:
> But Wait and Waker themselves are not necessarily the implementation detail.

Yes, they may not be the same thing. Whether we should expose `Waiter` and `Waker` depends on whether they might be used in other contexts and if they are self-explanatory abstractions.

Either keeping these structs hidden or not hidden is acceptable for me.

Reply:
Actually, I think `Waiter` and `Waker` are self-explanatory enough to be public. They are low-level primitives that are reasonable enough to be exposed by the framework.

For example, we can implement a very simple `sleep` based on these APIs (I know we could use `WaitQueue` or `Pauser` for the same purpose, but it's a bit complex for a simple `sleep`).